@Sigmanificient fixed the issue + assigned myself :-)

@felixsinger Here's the error (due to warnings treated as errors):
```
gcc -Wall -Werror -D_GNU_SOURCE -DVER_MAJOR=1 -DVER_MINOR=6    -O2 -DNDEBUG   -c -o commands.o commands.c
commands.c: In function 'make_command_links':
commands.c:153:49: error: '__builtin___snprintf_chk' output may be truncated before the last format character []
  153 |         snprintf(link_name, FILENAME_MAX, "%s/%s", path_to_bin, cmd->name);
      |                                                 ^
In file included from /nix/store/pz6jgrfqyl2il2785a75vmwdp9vypv0j-glibc-2.40-66-dev/include/stdio.h:970,
                 from commands.c:19:
In function 'snprintf',
    inlined from 'build_symlink_name' at commands.c:153:2,
    inlined from 'make_command_links' at commands.c:173:16:
/nix/store/pz6jgrfqyl2il2785a75vmwdp9vypv0j-glibc-2.40-66-dev/include/bits/stdio2.h:68:10: note: '__builtin___snprintf_chk' output 2 or more bytes (assuming 4097) into a destination of size 4096
   68 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   69 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   70 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
commands.c: In function 'clean_command_links':
commands.c:153:49: error: '__builtin___snprintf_chk' output may be truncated before the last format character []
  153 |         snprintf(link_name, FILENAME_MAX, "%s/%s", path_to_bin, cmd->name);
      |                                                 ^
In function 'snprintf',
    inlined from 'build_symlink_name' at commands.c:153:2,
    inlined from 'clean_command_links' at commands.c:204:16:
/nix/store/pz6jgrfqyl2il2785a75vmwdp9vypv0j-glibc-2.40-66-dev/include/bits/stdio2.h:68:10: note: '__builtin___snprintf_chk' output 2 or more bytes (assuming 4097) into a destination of size 4096
   68 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   69 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   70 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [<builtin>: commands.o] Error 1
```

--- a/pkgs/by-name/io/iotools/001-fix-werror-in-sprintf.patch
+++ b/pkgs/by-name/io/iotools/001-fix-werror-in-sprintf.patch
@@ -1,13 +1,27 @@
 diff --git a/commands.c b/commands.c
-index a83f1d5..b3f217a 100644
+index a28e6da..0f76ac7 100644
 --- a/commands.c
 +++ b/commands.c
-@@ -152,1 +152,7 @@ build_symlink_name(const char *path_to_bin, const struct cmd_info *cmd)
+@@ -20,6 +20,7 @@
+ #include <stdlib.h>
+ #include <string.h>
+ #include <unistd.h>
++#include <limits.h>
+ #include <errno.h>
+ #include "commands.h"
+ #include "platform.h"
+@@ -150,7 +151,13 @@ build_symlink_name(const char *path_to_bin, const struct cmd_info *cmd)
+ {
+ 	static char link_name[FILENAME_MAX];
+ 
 -	snprintf(link_name, FILENAME_MAX, "%s/%s", path_to_bin, cmd->name);
-+	int result = snprintf(link_name, FILENAME_MAX, "%s/%s", path_to_bin, cmd->name);
++	int result = snprintf(link_name, PATH_MAX, "%s/%s", path_to_bin, cmd->name);
 +
-+	if (result >= FILENAME_MAX) {
-+		link_name[FILENAME_MAX - 1] = '\0';
-+	} else if (result < 0) {
++	if (result >= PATH_MAX) {
++		link_name[PATH_MAX - 1] = '\0';
++		} else if (result < 0) {
 +		link_name[0] = '\0';
 +	}
+ 
+ 	return link_name;
+ }

--- a/pkgs/by-name/ps/psb_status/package.nix
+++ b/pkgs/by-name/ps/psb_status/package.nix
@@ -1,0 +1,44 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  bash,
+  iotools,
+  makeWrapper,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "psb_status";
+  version = "0-unstable-2024-10-10";
+
+  src = fetchFromGitHub {
+    owner = "mkopec";
+    repo = "psb_status";
+    rev = "be896832c53d6b0b70cf8a87f7ee46ad33deefc2";
+    hash = "sha256-4anPyjO8y3FgnYWa4bGFxI8Glk9srw/XF552tnixc8I=";
+  };
+
+  dontBuild = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -m755 psb_status.sh $out/bin/psb_status
+    wrapProgram $out/bin/psb_status \
+      --prefix PATH : ${lib.makeBinPath [ iotools ]}
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Script to check Platform Secure Boot enablement on Zen based AMD CPUs";
+    homepage = "https://github.com/mkopec/psb_status";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ phodina ];
+    platforms = [ "x86_64-linux" ];
+    mainProgram = "psb_status";
+  };
+})

